tools/shared.py

```diff
@@ -1388,6 +1388,7 @@ def print_compiler_stage(cmd):
   before executing it."""
   if '-v' in COMPILER_OPTS:
     print(' "%s" %s' % (cmd[0], ' '.join(cmd[1:])), file=sys.stderr)
+    sys.stderr.flush()
```

Comment:
why is this needed?

Reply:
Because otherwise the stderr will be buffered by default and the subprocess output will appear before this line reporting that we are about to run it.